src/osx/cocoa/combobox.mm

```diff
@@ -226,7 +231,17 @@ - (void)comboBoxSelectionDidChange:(NSNotification *)notification
 
 void wxNSComboBoxControl::InsertItem(int pos, const wxString& item)
 {
-    [m_comboBox insertItemWithObjectValue:wxCFStringRef( item , m_wxPeer->GetFont().GetEncoding() ).AsNSString() atIndex:pos];
+    if( m_wxPeer->HasFlag( wxCB_SORT ) )
+    {
+        NSArray *objectValues = m_comboBox.objectValues;
+        [m_comboBox insertItemWithObjectValue:wxCFStringRef( item , m_wxPeer->GetFont().GetEncoding() ).AsNSString()
```

Comment:
Could we please use a temporary variable to avoid repeating `wxCFStringRef( item , m_wxPeer->GetFont().GetEncoding() ).AsNSString()` 3 times in 6 lines of code?

Reply:
@vadz ,
Do you know how to get NSString from the wxString?

Calling this directly results in the compiler error.

Reply:
Use `NSString *`, as the hint shows.

Reply:
Oh, and don't do it like this, you'll end up with a dangling pointer to a temporary. Do
```
wxCFStringRef itemLabel(item);
NSString* const ns = itemLabel.AsNSString();
```

Reply:
@vadz 
> Oh, and don't do it like this, you'll end up with a dangling pointer to a temporary. Do
> 
> ```
> wxCFStringRef itemLabel(item);
> NSString* const ns = itemLabel.AsNSString();
> ```

This will break NSComparator compilation:

```
../src/osx/cocoa/combobox.mm:229:18: error: cannot initialize a variable of type 'NSComparator' (aka 'NSComparisonResult (^)(id _Nonnull, id
      _Nonnull)') with an rvalue of type 'NSComparisonResult (^)(id, id *)'
    NSComparator comp = ^(id obj1, id *obj2)
                 ^      ~~~~~~~~~~~~~~~~~~~~
1 error generated.

```

Reply:
> If I do it like this I will get a crash, because I will pass a pointer to comparator which will be compared to the actual string.

What could you possibly mean by this? This has nothing to do with the comparator...

Reply:
@vadz ,
I am sorry.
I did it like you proposed and there was no crush.

Everything is still works as it should.

Let's wait for the automatic build to finish and apply with closing the ticket.

Sorry about this. It was crushing with the backtrace starting at the comparator.